### PR TITLE
Adds StdLib exercises

### PR DESCRIPTION
**Pull request by rafaparadela**
This PR adds all the exercises taken from the version 1.

FYI: I have used [this script](https://gist.github.com/rafaparadela/b75f6a27b3712ed8738c) to automatise the migration.

@dialelo @raulraja Is good to merge?

### CODE REVIEW

File: site/content/src/main/exercises/stdlib/Constructors.scala [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+  /** primaryConstructorConstructors
+    *
+    * Primary Constructor
+    *
+    * In Java we have a no-args default constructor which is provided for every class which doesn't provide its own constructor methods. On a similar lines Primary Constructor in Scala is the kind-of default constructor in the way every class in Scala would have a Primary Constructor.
+    *
+    * The primary constructor spans the complete class definition i.e in the example below the age field gets initialized as part of the Primary Constructor of the Employee class.
+    *
+    *
+    * class Employee {
```
Comment (dialelo) [inline site/content/src/main/exercises/stdlib/Constructors.scala]: These code examples aren't enclosed in `{{{` and `}}}`

File: site/content/src/main/exercises/stdlib/Constructors.scala [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    * class Employee(val firstName:String,
+    * val lastName:String) {
+    *
+    * override def toString():String = {
+    * "First Name: " + firstName + " Last Name: " + lastName
+    * }
+    * }
+    *
+    *
+    * #### Auxiliary Constructor
```
Comment (dialelo) [inline site/content/src/main/exercises/stdlib/Constructors.scala]: Scaladoc uses `=` symbols on both sides for marking headers, we may want to use that syntax.

File: site/content/src/main/exercises/stdlib/EmptyValues.scala [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+    *
+    * ==Nil==
+    *
+    * [Nil](http://www.scala-lang.org/api/current/index.html#scala.collection.immutable.Nil$) is just an empty list, exactly like the result of `List()`. It is of type `List[Nothing]`. And since we know there are no instances of Nothing, we now have a list that is statically verifiable as empty. Nice to have. 
+    *
+    * An empty list can be represented by another nothing value: *Nil*
+    */
+  def emptyValuesEmptyValues(res0: String) {
+    assert(List() === res0)
+
```
Comment (dialelo) [inline site/content/src/main/exercises/stdlib/EmptyValues.scala]: All the exercises have a couple of empty lines in the end, can we get rid of it?

File: site/content/src/main/exercises/stdlib/Extractors.scala [hunk 4]
```diff
@@ -0,0 +1,236 @@
+package stdlib
+
+    import org.scalatest._
+
+    /** Extractors
+      *
+      * Extractors Description
+      */
+
+    object Extractors extends FlatSpec with Matchers with exercise.Section {
```
Comment (dialelo) [inline site/content/src/main/exercises/stdlib/Extractors.scala]: It'd be nice if Scalariform formatted these files.

File: site/content/src/main/exercises/stdlib/Asserts.scala [hunk 5]
```diff
@@ -0,0 +1,75 @@
+package stdlib
+
+import org.scalatest._
+
+/** Asserts
+  *
+  * Asserts Description
+  */
+
```
Comment (raulraja) [inline site/content/src/main/exercises/stdlib/Asserts.scala]: Is there a way to remove this empty line?

File: site/content/src/main/exercises/stdlib/Asserts.scala [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+    * result should be (3)    // cannot customize equality, so fastest to compile
+    * result shouldEqual 3    // can customize equality, no parentheses required
+    * result shouldBe 3       // cannot customize equality, so fastest to compile, no parentheses required
+    * }}}
+    *
+    * Come on, your turn: true and false values can be compared with should matchers
+    */
+  def scalaTestAsserts(res0: String) {
+    true should be(res0)
+
```
Comment (raulraja) [inline site/content/src/main/exercises/stdlib/Asserts.scala]: Is there a way to remove these empty spaces?

**Comment by raulraja** [comment]
Rafa this is awesome. Agreed with @dialelo comments. If we can solve those small formatting issues and format everything with Scalariform this is a great step forward and should be good to merge. LGTM

File: site/content/src/main/exercises/stdlib/Constructors.scala (hunk 1 above)
Comment (rafaparadela) [inline site/content/src/main/exercises/stdlib/Constructors.scala]: Done! In the original source this code wasn't within  `triple backticks`

File: site/content/src/main/exercises/stdlib/Constructors.scala (hunk 2 above)
Comment (rafaparadela) [inline site/content/src/main/exercises/stdlib/Constructors.scala]: Replaced every occurrence `####` to `==whatever==`.

File: site/content/src/main/exercises/stdlib/EmptyValues.scala (hunk 3 above)
Comment (rafaparadela) [inline site/content/src/main/exercises/stdlib/EmptyValues.scala]: Removed

File: site/content/src/main/exercises/stdlib/Extractors.scala (hunk 4 above)
Comment (rafaparadela) [inline site/content/src/main/exercises/stdlib/Extractors.scala]: Scalariform doesn't work properly with block code inside of scaladocs :-(

File: site/content/src/main/exercises/stdlib/Asserts.scala (hunk 5 above)
Comment (rafaparadela) [inline site/content/src/main/exercises/stdlib/Asserts.scala]: This empty lines were introduced in my script. That's fixed now.

File: site/content/src/main/exercises/stdlib/Asserts.scala (hunk 6 above)
Comment (rafaparadela) [inline site/content/src/main/exercises/stdlib/Asserts.scala]: This empty lines were introduced in my script. That's fixed now.

**Pull request merged**